send_rx_functions.php

```diff
@@ -0,0 +1,193 @@
+ *   The file path to save the file.
+ *
+ * @return bool
+ *   TRUE if success, FALSE otherwise.
+ */
+function send_rx_generate_pdf_file($contents, $file_path) {
+    // TODO.
+}
+
+/**
+ * Updates a data entry field value.
+ *
+ * @param int $project_id
+ *   Data entry project ID.
+ * @param int $event_id
+ *   Data entry event ID.
+ * @param int $record_id
+ *   Data entry record ID.
+ * @param string $field_name
+ *   Machine name of the field to be updated.
+ * @param string $value
+ *   The value to be saved.
+ *
+ * @return bool
+ *   TRUE if success, FALSE otherwise.
+ */
+function send_rx_update_record_field($project_id, $event_id, $record_id, $field_name, $value) {
+    // TODO.
+}
+
+/**
+ * Gets data entry record information.
+ *
+ * If more than one event is fetched, it returns the first one.
+ *
+ * @param int $project_id
+ *   Data entry project ID.
+ * @param int $record_id
+ *   Data entry record ID.
+ * @param int $event_id
+ *   (optional) Data entry event ID.
+ *
+ * @return array|bool
+ *   Data entry record information array. FALSE if failure.
+ */
+function send_rx_get_record_data($project_id, $record_id, $event_id = NULL) {
+    $data = REDCap::getData($project_id, 'array', $record_id, NULL, $event_id);
+    if (empty($data[$record_id])) {
+        return FALSE;
+    }
+
+    if ($event_id) {
+        return $data[$record_id][$event_id];
+    }
+
+    return reset($data[$record_id]);
+}
+
+/**
+ * Gets repeat instrument instances information from a given data entry record.
+ *
+ * If more than one event is fetched, it returns the first one.
+ *
+ * @param int $project_id
+ *   Data entry project ID.
+ * @param int $record_id
+ *   Data entry record ID.
+ * @param int $event_id
+ *   (optional) Data entry event ID.
+ *
+ * @return array|bool
+ *   Array containing repeat instrument instances information. FALSE if failure.
+ */
+function send_rx_get_repeat_instrument_instances($project_id, $record_id, $instrument_name, $event_id = NULL) {
+    $data = REDCap::getData($project_id, 'array', $record_id, NULL, $event_id);
+    if (empty($data[$record_id]['repeat_instances'])) {
+        return FALSE;
+    }
+
+    $data = $event_id ? $data[$record_id]['repeat_instances'][$event_id] : reset($data['repeat_instances'][$record_id]);
+    if (empty($data[$instrument_name])) {
+        return FALSE;
+    }
+
+    return $data[$instrument_name];
+}
+
+/**
+ * Locks for updates the given instruments of the given data entry record.
+ *
+ * @param int $project_id
+ *   Data entry project ID.
+ * @param int $record_id
+ *   Data entry record ID.
+ * @param int $instruments
+ *   (optional) Array of instruments names. Leave blank to block all instruments.
+ * @param int $event_id
+ *   (optional) Data entry event ID. Leave blank to block all events.
+ *
+ * @return bool
+ *   TRUE if success, FALSE otherwise.
+ */
+function send_rx_lock_instruments($project_id, $record_id, $instruments = NULL, $event_id = NULL) {
```

Comment:
Yes. I am creating a separate class for locking forms and events. So, I guess extra parameter need not be added.